#10 - Post about how to configure maven repo

genryxy opened the pull request:
Closes #10 
Add post about how to configure maven repository and Maven project to use it.

genryxy commented:
@olegmoz @olenagerasimova please review

olegmoz commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
```diff
@@ -0,0 +1,122 @@
+
+## Introduction
+
+Nowadays Apache Maven is [one of the most popular](https://www.baeldung.com/java-in-2019#build-tools-adoption) build management tool for Java. Many companies that do Java development have private Maven repositories, where they store all the artifacts. For example, the team does not work in open source and has some private artifacts. This team needs to share easily and quickly among themselves and ...
```
@genryxy the start of the blog post feels like an introduction of some scientific article. I think it should be more lively, addressing an actual person who may need Artipie in team/company. Your reader already did choose Java and is using Maven as build tool, you do not need to explain why Java/Maven are popular. Rather I'd suggest you to state problem our client may solve using Artipie: breaking some big repository into multiple repositories possibly developed by different people/teams. That looks like case when team/company going to need and artifact manager. Other one would be creating local cache for build artifacts from Maven Central, it might be the theme for another blog post.

olegmoz commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
```diff
@@ -0,0 +1,122 @@
+
+If you have completed all the instructions correctly now you should be able to use the repository: deploy and install Maven artifacts.
+
+## Configuring project to use your Artipie maven repository
```
@genryxy I'd suggest to move this section earlier, before authentication. So we start first with running the server and adding it to `settings.xml` so it works. Then complicate things further with restricting access

olegmoz commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
```diff
@@ -0,0 +1,122 @@
+[documentation section](https://github.com/artipie/artipie/tree/master/examples/maven)
+for more details.
+
+## Conclusion
+
+In this blog post I explained how to set up own Maven repository with Artipie and how to
+configure a Maven project to use it. 
```
@genryxy this section again feels too formal, like something from an scientific article, I'd suggest to make it more informal. Maybe it feels that way because of labeled sections, blog post are expected to be less formal and be more like a story, not a document

olegmoz reviewed (changes requested):
@genryxy thank you for this post, please see some comments

genryxy commented:
@olegmoz thanks, refactored introduction according to the comment. Moved the case with authentication to one section near the end. Add one more sentence in conclusion.

olegmoz commented:
@yegor256 could you please review this blog post draft?

yegor256 commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
````diff
@@ -0,0 +1,136 @@
+    path: /var/artipie/configs
+```
+
+After that you can install and run Artipie with docker:
````
@genryxy what is "docker"? Maybe a cross link would be helpful? Also, it's Docker, not docker.

yegor256 commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
````diff
@@ -0,0 +1,136 @@
+
+```bash
+$ docker run -d --name artipie \
+             -v $(pwd)/artipie.yaml:/etc/artipie.yml \
````
@genryxy why so many leading spaces?

genryxy replied:
@yegor256 to align the flags on the same level. Is it better to remove extra spaces?

yegor256 commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+date: 2020-11-06
+title: "Using Artipie as a maven repository"
```
@genryxy it's Maven, not maven

yegor256 commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
```diff
@@ -0,0 +1,136 @@
+layout: post
+date: 2020-11-06
+title: "Using Artipie as a maven repository"
+author: Alexander Krasnov
```
@genryxy we need your GitHub name here (see how it's done in other blog posts)

yegor256 commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
```diff
@@ -0,0 +1,136 @@
+[documentation section](https://github.com/artipie/artipie/tree/master/examples/maven)
+for more details.
+
+## Conclusion
```
@genryxy I would get rid of all section headers in such a small blog post. They are not necessary and only makes the text noisy

yegor256 commented:
@olegmoz @genryxy I posted a few comments above

genryxy commented:
@yegor256 thanks, corrected. I replaced large headers with lines with bold text.

yegor256 commented:
@genryxy I would suggest getting rid of headers at all. They are not needed if the size of each "section" is just one paragraph. Headers are required for large texts, where the reader can easily get lost. In this small blog post a reader can't get lost)

genryxy commented:
@yegor256 thanks, I removed all headers except one to separate somehow the part where it is written about authentication

yegor256 commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
````diff
@@ -0,0 +1,124 @@
+```
+If you have completed all the instructions correctly now you should be able to use the repository: deploy and install Maven artifacts.
+
+**Configure Artipie with authentication**
````
@genryxy this time it's better to make it a header, I think. Using BOLD text instead of native Markdown formatting is a bad practice

genryxy commented:
@yegor256 thanks, corrected

yegor256 commented:
@olegmoz @genryxy looks good to me now

olenagerasimova commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
```diff
@@ -0,0 +1,124 @@
+
+Don't forget to run Artipie server with Docker and to update `pom.xml` of your project as shown above. If all the steps were done correctly you should be able to use the repository with authentication: deploy and install Maven artifacts.
+
+Apart from demonstrated possibilities, Artipie Maven supports: group repositories, proxies and user groups.Check out the corresponding
```
@genryxy space is missing after dot

olenagerasimova commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
```diff
@@ -0,0 +1,124 @@
+[documentation section](https://github.com/artipie/artipie/tree/master/examples/maven)
+for more details.
+
+In this blog post I explained how to set up own Maven repository with Artipie and how to
```
@genryxy I'm not sure the phrase "to set up own Maven" is grammatically correct, maybe we can replace "own" with "private" or "your own"?

olenagerasimova commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
````diff
@@ -0,0 +1,124 @@
+             artipie/artipie:latest
+```
+
+In the previous step, we created a file with settings for Artipie. Now let's see what the configuration file for the repository should look like. For this, we need to create `configs` directory and the following `my-maven.yaml` file inside:
````
@genryxy 
> "we created a file"

in this phrase "a" is definitely not required as we are talking about the concrete file, not some random file

olenagerasimova commented on _posts/2020/11/2020-11-06-configure-artipie-maven-repo.md:
````diff
@@ -0,0 +1,124 @@
+    path: _credentials.yml
+```
+
+Also, it's necessary to create a `_credentials.yml` file in the `configs` directory. Each node is a user name. The supported types of password can be found [here](https://github.com/artipie/artipie/blob/700eb89352126e6f1bd12a0d3ff668abf2b44048/README.md#multitenancy) The file with credentials should look like this:
````
@genryxy I'd suggest to omit "a" in 
> it's necessary to create a `_credentials.yml`

olenagerasimova reviewed (changes requested):
@genryxy several minor suggestions, please, take a look. Also, I'd suggest to keep line length not too long, single line brakes does not affect markdown and it's easier to read the text this way.

genryxy commented:
@olenagerasimova thanks, corrected

olenagerasimova reviewed (approved):
@genryxy thanks
@olegmoz looks good to me
